packages/mco/components/create-dr-policy/add-s3-bucket-details/s3-bucket-details-form.tsx

```diff
@@ -0,0 +1,230 @@
+import * as React from 'react';
+import {
+  Button,
+  Checkbox,
+  Divider,
+  ExpandableSection,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { EyeIcon, EyeSlashIcon } from '@patternfly/react-icons';
+import { DRPolicyActionType, ManagedClusterInfoType } from '../utils/reducer';
+
+export type S3Details = {
+  clusterName: string;
+  bucketName: string;
+  endpoint: string;
+  accessKeyId: string;
+  secretKey: string;
+  region: string;
+  s3ProfileName: string;
+};
+
+type ClusterS3BucketDetailsFormProps = {
+  selectedClusters: ManagedClusterInfoType[];
+  cluster1Details: S3Details;
+  cluster2Details: S3Details;
+  useSameConnection: boolean;
+  dispatch: React.Dispatch<any>;
+};
+
+export const ClusterS3BucketDetailsForm: React.FC<
+  ClusterS3BucketDetailsFormProps
+> = ({
+  selectedClusters,
+  cluster1Details,
+  cluster2Details,
+  useSameConnection,
+  dispatch,
+}) => {
+  const [expanded1, setExpanded1] = React.useState(true);
+  const [expanded2, setExpanded2] = React.useState(true);
+  const [show1, setShow1] = React.useState(false);
+  const [show2, setShow2] = React.useState(false);
+
+  const [errors1, setErrors1] = React.useState<
+    Partial<Record<keyof S3Details, string>>
+  >({});
+  const [errors2, setErrors2] = React.useState<
+    Partial<Record<keyof S3Details, string>>
+  >({});
+
+  const name1 = selectedClusters[0]?.metadata?.name || 'cluster-1';
+  const name2 = selectedClusters[1]?.metadata?.name || 'cluster-2';
+
+  React.useEffect(() => {
+    if (useSameConnection) {
+      dispatch({
+        type: DRPolicyActionType.SET_CLUSTER2_S3_DETAILS,
+        payload: { ...cluster1Details, clusterName: name2 },
+      });
+      setErrors2({});
+    }
+  }, [useSameConnection, cluster1Details, dispatch, name2]);
+
+  const validate = (cluster: 1 | 2, details: S3Details) => {
+    const errs: Partial<Record<keyof S3Details, string>> = {};
+    if (!details.bucketName.trim()) errs.bucketName = 'Bucket name is required';
+    if (!details.endpoint.trim()) errs.endpoint = 'Endpoint is required';
+    if (!details.accessKeyId.trim())
+      errs.accessKeyId = 'Access key ID is required';
+    if (!details.secretKey.trim()) errs.secretKey = 'Secret key is required';
+    if (!details.region.trim()) errs.region = 'Region is required';
+    if (!details.s3ProfileName.trim())
+      errs.s3ProfileName = 'S3 profile name is required';
+    cluster === 1 ? setErrors1(errs) : setErrors2(errs);
+  };
```

Comment:
u don't have to manage error state at all... u are only showing "Required field" error that's all... use common text "This field is required" for all, u are anyway displaying this error right below the corresponding TextInput element, there is no need to display the field name in the error message at all.
```suggestion
  const name1 = selectedClusters[0]?.metadata?.name || 'cluster-1';
  const name2 = selectedClusters[1]?.metadata?.name || 'cluster-2';

  React.useEffect(() => {
    if (useSameConnection) {
      dispatch({
        type: DRPolicyActionType.SET_CLUSTER2_S3_DETAILS,
        payload: { ...cluster1Details, clusterName: name2 },
      });
      setErrors2({});
    }
  }, [useSameConnection, cluster1Details, dispatch, name2]);
```

Reply:
@SanjalKatiyar It was in the mockups earlier, it went through multiple redesigns and ended up as the current one. The S3 details had error details underneath them but did not have the specific field name.

https://www.figma.com/design/ph6iiSTQD6bxEyw0rg0x3N/DR-for-Non--ODF-cluster--4.20-?node-id=1-4&p=f&t=t6BrsdP4ktyTTkUO-0